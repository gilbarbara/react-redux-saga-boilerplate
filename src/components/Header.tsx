import { useDispatch } from 'react-redux';
import styled from '@emotion/styled';
import { Container, Icon, responsive, Text } from '@gilbarbara/components';

import { appColor, headerHeight } from '~/modules/theme';

import { logOut } from '~/actions';

import Logo from '~/components/Logo';

const HeaderWrapper = styled.header`
  background-color: #113740;
  height: ${headerHeight}px;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 200;

  &:before {
    background-color: ${appColor};
    bottom: 0;
    content: '';
    height: 2px;
    left: 0;
    position: absolute;
    right: 0;
  }
`;

const Logout = styled.button`
  align-items: center;
  color: #fff;
  display: flex;
  font-size: 14px;

  ${responsive({ lg: { fontSize: '16px' } })};

  span {
    display: inline-block;
    text-transform: uppercase;
  }
`;

export default function Header() {
  const dispatch = useDispatch();

  const handleClickLogout = () => {
    dispatch(logOut());
  };

  return (
    <HeaderWrapper data-component-name="Header">
      <Container direction="row" justify="space-between" padding="md">
        <Logo />
        <Logout data-component-name="Logout" onClick={handleClickLogout}>
          <Text>logout</Text>
          <Icon ml="xs" name="sign-out" />
        </Logout>
      </Container>
    </HeaderWrapper>
  );
}
