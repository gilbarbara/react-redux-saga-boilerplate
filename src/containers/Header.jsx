import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import rgba from 'polished/lib/color/rgba';
import { appColor, headerHeight } from 'modules/theme';

import { logOut } from 'actions/index';

import { Container, utils } from 'styled-minimal';
import Icon from 'components/Icon';
import Logo from 'components/Logo';

const { responsive, spacer } = utils;

const HeaderWrapper = styled.header`
  background-color: ${rgba(appColor, 0.9)};
  height: ${headerHeight}px;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 200;

  &:before {
    background-color: ${appColor};
    bottom: 0;
    content: '';
    height: 0.2rem;
    left: 0;
    position: absolute;
    right: 0;
  }
`;

const HeaderLogo = styled(Logo)`
  svg {
    width: 5rem;
  }
`;

const StyledContainer = styled(Container)`
  align-items: center;
  display: flex;
  flex-wrap: wrap;
  height: 100%;
  justify-content: space-between;
  padding-bottom: ${spacer(2)};
  padding-top: ${spacer(2)};
`;

const Logout = styled.button`
  align-items: center;
  color: #333;
  display: flex;
  font-size: 1.3rem;
  padding: ${spacer(2)};

  ${responsive({ lg: 'font-size: 1.6rem;' })}; /* stylelint-disable-line */

  &.active {
    color: #000;
  }

  span {
    display: inline-block;
    margin-right: 0.4rem;
    text-transform: uppercase;
  }
`;

export default class Header extends React.PureComponent {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
  };

  handleClickLogout = () => {
    const { dispatch } = this.props;

    dispatch(logOut());
  };

  render() {
    return (
      <HeaderWrapper>
        <StyledContainer>
          <HeaderLogo />
          <Logout onClick={this.handleClickLogout}>
            <span>logout</span>
            <Icon name="sign-out" color="#333" width={16} />
          </Logout>
        </StyledContainer>
      </HeaderWrapper>
    );
  }
}
