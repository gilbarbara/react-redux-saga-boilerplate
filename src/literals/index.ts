import { keyMirror } from '@gilbarbara/helpers';

import { Status } from '~/types';

export const ActionTypes = keyMirror({
  ALERTS_HIDE: undefined,
  ALERTS_SHOW: undefined,
  GITHUB_GET_REPOS_REQUEST: undefined,
  GITHUB_GET_REPOS_SUCCESS: undefined,
  GITHUB_GET_REPOS_FAILURE: undefined,
  SET_APP_OPTIONS: undefined,
  USER_LOGIN_REQUEST: undefined,
  USER_LOGIN_SUCCESS: undefined,
  USER_LOGIN_FAILURE: undefined,
  USER_LOGOUT_REQUEST: undefined,
  USER_LOGOUT_SUCCESS: undefined,
  USER_LOGOUT_FAILURE: undefined,
});

export const STATUS: Status = {
  IDLE: 'idle',
  RUNNING: 'running',
  READY: 'ready',
  SUCCESS: 'success',
  ERROR: 'error',
};
