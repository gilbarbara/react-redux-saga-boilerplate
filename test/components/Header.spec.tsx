import React from 'react';
import { fireEvent, render, screen } from 'test-utils';

import { ActionTypes } from '~/literals';

import Header from '~/components/Header';

const mockDispatch = vi.fn();

describe('Header', () => {
  it('should render properly', () => {
    render(<Header />);

    expect(screen.getByRole('banner')).toMatchSnapshot();
  });

  it('should handle clicks', async () => {
    render(<Header />, { mockDispatch });
    fireEvent.click(screen.getByRole('button')!);

    expect(mockDispatch).toHaveBeenCalledWith({
      type: ActionTypes.USER_LOGOUT_REQUEST,
    });
  });
});
