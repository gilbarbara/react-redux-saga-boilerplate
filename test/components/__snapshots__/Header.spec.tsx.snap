// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > should render properly 1`] = `
<header
  class="css-157to9t"
  data-component-name="Header"
>
  <div
    class="css-1ejvhcb"
    data-component-name="Container"
  >
    <div
      class="css-11mttn6"
    >
      <svg
        src="/media/brand/icon.svg"
      />
    </div>
    <button
      class="css-17rt720"
      data-component-name="Logout"
    >
      <span
        class="css-lo0v0o"
        data-component-name="Text"
      >
        logout
      </span>
      <span
        class="css-10mt42n"
        data-component-name="Icon"
        name="sign-out"
      >
        <svg
          height="16"
          src="https://files.gilbarbara.dev/icons/sign-out.svg"
          title="sign-out"
          width="16"
        />
      </span>
    </button>
  </div>
</header>
`;
